src/libraries/Microsoft.PowerFx.Core/Syntax/Nodes/TypeLiteralNode.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Syntax.SourceInformation;
+
+namespace Microsoft.PowerFx.Syntax
+{
+    // CallNode :: Type ( TypeLiteralNode )
+    // CallNode :: IsType ( TypeLiteralNode )
+    // CallNode :: AsType ( Expr, TypeLiteralNode )
+
+    public sealed class TypeLiteralNode : TexlNode
+    {
+        internal DType Type { get; }
```

Comment:
>DType [](http://example.com/codeflow?start=17&length=5)

Should this node hold a dtype, or something else?

One thing to consider, if we allow named types in the future, can you reference those within a type literal node, and if we delete/rename a named type elsewhere, would that require reparsing all it's references to rebuild this DType?

Reply:
Removed DType temporarily just to focus on getting the AST Node changes through.